src/echoclient/EchoClient.php

```diff
@@ -62,51 +65,45 @@ function __construct()
     }
 
     /**
-     * Add an event to echo
+     * Adds a single event to echo
      *
      * @param string $class
      * @param string $source
      * @param array $props
      * @param string|null $userId
+     * @param string|null $timestamp
      * @return bool True if successful, else false
      */
-    public function createEvent($class, $source, array $props = array(), $userId = null)
+    public function createEvent($class, $source, array $props = array(), $userId = null, $timestamp = null)
     {
         $class = ECHO_CLASS_PREFIX . $class;
-        $baseUrl = $this->getBaseUrl();
+        $eventJson = $this->getEventJson($class, $source, $props, $userId, $timestamp);
 
-        if (!$baseUrl) {
-            // fail silently when creating events, should not stop user interaction as echo events are collected on a best-endeavours basis
-            $this->getLogger()->warning('Echo server is not defined (missing ECHO_HOST define), not sending event - ' . $class,
-                $props);
-            return false;
-        }
-
-        $eventUrl = $baseUrl . '/events';
-        $eventJson = $this->getEventJson($class, $source, $props, $userId);
+        return $this->sendJsonEventDataToEcho($eventJson);
+    }
 
-        try {
-            $client = $this->getHttpClient();
-            $request = $client->post($eventUrl, $this->getHeaders(), $eventJson, array('connect_timeout' => 2));
-            $response = $request->send();
+    /**
+     * Adds multiple events to echo
+     *
+     * @param EchoEvent[] $events An array of EchoEvent objects
+     * @return bool True if successful, else false
+     */
+    public function createEvents(array $events)
+    {
+        if (count($events) > self::ECHO_MAX_BATCH_EVENTS) {
+            $this->getLogger()->warning('Batch contains more than ' . self::ECHO_MAX_BATCH_EVENTS . ' events');
+            throw new \Exception("Batch of events exceeds the maximum allowed size");
```

Comment:
Is there a check for > allowed size in MB?

Reply:
There is now - below.